### DISCUSSION TITLE
Array of References

### DISCUSSION THREAD

**Discussion by mashaalmemon** [xaviergonz/mobx-keystone#301] [Q&A]
What is the proper way to store an array of references? For my references I setup something like this:

```typescript
@model("NTodo")
export class NTodo
  extends ExtendedModel(BaseNModel, {
    userListRefs: Ref<NUser>[]
  })
  implements Partial<ITodo>
{

  /**
   * Get list of users
   */
  @computed
  public get userList(): IUser[] {
    let userList: IUser[] = [];
    for (let i: number = 0; i < this.userListRefs.length; i++) {
      // @ts-ignore
      walletList.push(userList[i].current));
    }

    return walletList;
  }

}
```

Does something like this equate to an observable array, or do I need to do things differently?

**Comment by xaviergonz** [comment]
```ts
@model("NTodo")
export class NTodo
  extends ExtendedModel(BaseNModel, {
    userListRefs: prop<Ref<NUser>[]>() // just this should be enough, might want to add a default `() => []`
  })
  implements Partial<ITodo>
{

  // this method is optional
  @computed
  public get userList(): IUser[] {
    // return all refs that are not broken
    // if we use .current we risk this method to throw if one of the refs is broken (might be
    // what you want though)
    return this.userListRefs.map(r => r.maybeCurrent).filter(r => r !== undefined)
  }

}
```

> Does something like this equate to an observable array, or do I need to do things differently?

I'm not quite sure what you mean, all arrays in mobx-keystone are observable arrays, but an array being observable has nothing to do with references 🤷‍♂️

**Reply by mashaalmemon** [reply]
Got it. Perfect.

Re `.current` vs `.maybeCurrent` got it. 

Ok, all arrays in box-keystone are observable arrays. Perfect. So if I push an item onto the array or remove an item from the array, that will trigger anything looking at the values of the array to update then.

That's exactly what I need. Wanted to make sure of that.
